File: codewars/python/7kyu/growth_of_a_population.py
def nb_year(p0, percent, aug, p):
    years = 0
    while p0 < p:
        p0 = int(p0 + p0 * percent/100 + aug)
        years += 1
    return years


if __name__ == '__main__':
    print(nb_year(1000, 2.0, 50, 1214))
    print(nb_year(1500000, 0.0, 10000, 2000000))

File: codewars/go/7kyu/growth_of_a_population.go
package main

import "fmt"

func main() {
	fmt.Println(NbYear(1000, 2.0, 50, 1214))
	fmt.Println(NbYear(1500000, 0.0, 10000, 2000000))
}

func NbYear(p0 int, percent float64, aug int, p int) int {
	years := 0
	for p0 < p {
		p0 = p0 + int(float64(p0)*percent/100) + aug
		years++
	}
	return years
}
